test/spec/Garn/CodeGenSpec.hs

```diff
@@ -2,161 +2,372 @@
+          `shouldReturn` unindent
+            [i|
+              import { mkPackage } from "path/to/garn/root/../../package.ts";
+              import { nixRaw } from "path/to/garn/root/../../nix.ts";
+
+              export const some_pkg = mkPackage(
+                nixRaw`pkgs.nested-1.nested-2.some-pkg`,
+                "",
+              );
+            |]
+
+    testWritePkgFiles
+      "sanitizes package names"
+      ( ("a-package" ~> Derivation {description = Nothing, path = "pkgs.a-package"})
+          <> ("a/b@c" ~> Derivation {description = Nothing, path = "pkgs.a/b@c"})
+          <> ("catch" ~> Derivation {description = Nothing, path = "pkgs.catch"})
+          <> ("instanceOf" ~> Derivation {description = Nothing, path = "pkgs.instanceOf"})
+          <> ("super" ~> Derivation {description = Nothing, path = "pkgs.super"})
+          <> ("void" ~> Derivation {description = Nothing, path = "pkgs.void"})
+      )
+      $ \readFile -> do
+        readFile "mod.ts"
+          `shouldReturn` unindent
+            [i|
+              import { mkPackage } from "path/to/garn/root/package.ts";
+              import { nixRaw } from "path/to/garn/root/nix.ts";
+
+              export const a_b_c = mkPackage(
+                nixRaw`pkgs.a/b@c`,
+                "",
+              );
+
+              export const a_package = mkPackage(
+                nixRaw`pkgs.a-package`,
+                "",
+              );
+
+              export const catch_ = mkPackage(
+                nixRaw`pkgs.catch`,
+                "",
+              );
+
+              export const instanceOf_ = mkPackage(
+                nixRaw`pkgs.instanceOf`,
+                "",
+              );
+
+              export const super_ = mkPackage(
+                nixRaw`pkgs.super`,
+                "",
+              );
+
+              export const void_ = mkPackage(
+                nixRaw`pkgs.void`,
+                "",
+              );
+            |]
+    testWritePkgFiles
+      "removes conflicts due to sanitization"
+      ( ("a-package" ~> Derivation {description = Nothing, path = "pkgs.a-package"})
+          <> ("a_package" ~> Derivation {description = Nothing, path = "pkgs.a_package"})
+          <> ("a/package" ~> Derivation {description = Nothing, path = "pkgs.a/package"})
+      )
+      $ \readFile -> do
+        readFile "mod.ts"
+          `shouldReturn` unindent
+            [i|
+              import { mkPackage } from "path/to/garn/root/package.ts";
+              import { nixRaw } from "path/to/garn/root/nix.ts";
+
+              export const a_package = mkPackage(
+                nixRaw`pkgs.a_package`,
+                "",
+              );
+            |]
+
+testScanPackages :: String -> String -> String -> Map String PkgInfo -> SpecWith ()
+testScanPackages description pkgs pkgSpec expectedOutput =
+  it description $ do
+    withModifiedEnvironment [("NIX_CONFIG", "experimental-features =")] $ do
```

Comment:
This could also go into an [aroundAll_](https://www.stackage.org/haddock/lts-21.20/hspec-2.10.10/Test-Hspec.html#v:aroundAll_). That would mean it gets executed once for all the tests in this `Spec`.

Reply:
Very much TIOLI! It's just that you mentioned you don't like this here.